flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -85,23 +86,31 @@
 
 	public KubernetesResourceManagerDriver(
 			Configuration flinkConfig,
-			FlinkKubeClient kubeClient,
 			KubernetesResourceManagerDriverConfiguration configuration) {
 		super(flinkConfig, GlobalConfiguration.loadConfiguration());
-
 		this.clusterId = Preconditions.checkNotNull(configuration.getClusterId());
 		this.podCreationRetryInterval = Preconditions.checkNotNull(configuration.getPodCreationRetryInterval());
-		this.kubeClient = Preconditions.checkNotNull(kubeClient);
 		this.requestResourceFutures = new HashMap<>();
 		this.podCreationCoolDown = FutureUtils.completedVoidFuture();
 	}
 
+	public KubernetesResourceManagerDriver(
+			Configuration flinkConfig,
+			FlinkKubeClient kubeClient,
+			KubernetesResourceManagerDriverConfiguration configuration) {
+		this(flinkConfig, configuration);
+		this.kubeClient = Preconditions.checkNotNull(kubeClient);
+	}
+
 	// ------------------------------------------------------------------------
 	//  ResourceManagerDriver
 	// ------------------------------------------------------------------------
 
 	@Override
 	protected void initializeInternal() throws Exception {
+		if (kubeClient == null) {
+			kubeClient = KubeClientFactory.fromConfiguration(flinkConfig, getIoExecutor());
```

Comment:
Currently this line of code cannot be covered by the unit tests, due to usage of a different constructor.